### ISSUE TITLE
[HOLD] Back button on Android devices while in chat screen exits application instead of navigating back to the chat list

### ISSUE THREAD

**Issue by jasperhuangg** [Expensify/Expensify.cash#1655]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding!
___

**Platform - version:**  
Android - all versions

**Action Performed (reproducible steps):**
1. Open a chat.
2. Use the back button (or the swipe from edge gesture if you have it set up).

**Expected Result:**
It should navigate back to the Chats list (like in Slack, or any other messaging app). Doesn't matter which, they trigger the same system back event.

**Actual Result:**
It exits the application.

**Notes/Photos/Videos:** 
This provides a very strange experience for users. Normally on Android, when a back button exists in an application's UI, the native UI's back button is expected to have the same behavior. There are many times where I've been using the app and I've hit the native back button expecting it to go back to my chats, but it just exits the app altogether. 

Should be a relatively easy fix with https://reactnative.dev/docs/backhandler

https://user-images.githubusercontent.com/31285285/110271354-d6627a00-8002-11eb-8b0e-f28a30844447.mp4

**Logs - JS/Android/iOS (if applicable)**:
N/A

**Comment by MelvinBot** [comment]
Triggered auto assignment to @michelleniemi (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @AndrewGable (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @NicMendonca (`External`), see https://stackoverflow.com/c/expensify/questions/7703 for more details.

**Comment by NicMendonca** [comment]
Going to re-assign for someone else to post in UpWork (I am going OOO for a week and won't be able to manage this)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @mallenexpensify (`External`), see https://stackoverflow.com/c/expensify/questions/7703 for more details.

**Comment by mallenexpensify** [comment]
@jasperhuangg couple questions
1. Was there a reason this was created in e.cash repo instead of E/E? 
2. Is this the same (or similar-enough) issue? https://github.com/Expensify/Expensify/issues/157248  Do you think once that is fixed that it _might_ fix this too?  We could put this on hold to wait to find out or.. not.

**Comment by marcaaron** [comment]
Not sure what device this is. But can confirm this does not happen in Android emulator with the latest version of `@react-navigation/drawer`. I agree with @mallenexpensify that we should put this on hold to see if the issue is resolved.

**Comment by AndrewGable** [comment]
@jasperhuangg - Assigning you to answer the questions above. Thanks!

**Comment by jasperhuangg** [comment]
Hey guys! So sorry for the late reply!! I was OOO for for a few days last week and this completely got lost in my notifications 😅

> Was there a reason this was created in e.cash repo instead of E/E?

Not in particular, let me know if I was supposed to create it elsewhere! I figured if I created it in E.cash then it would get exported.

> Is this the same (or similar-enough) issue? Expensify/Expensify#157248 Do you think once that is fixed that it might fix this too? We could put this on hold to wait to find out or.. not.

Looks like these issues are actually unrelated. Previously when I was testing we didn't have the right back-button behavior defined, so it would just exit the app entirely and take me to my Android home. I just tested it on my own Android device and it seems like that isn't the case anymore, so I'll close this out since it doesn't seem like an issue anymore!
